packages/common/src/api/tan-query/wallets/useAudioBalance.ts

```diff
@@ -220,11 +232,81 @@ export const useAudioBalance = (options: UseAudioBalanceOptions = {}) => {
   // Together they are the total balance
   const totalBalance = AUDIO(accountBalance + connectedWalletsBalance).value
   const isLoading = isAccountBalanceLoading || isConnectedWalletsBalanceLoading
-
+  const isError =
+    isConnectedWalletsError ||
+    accountBalances.some((balanceRes) => balanceRes.isError) ||
+    connectedWalletsBalances.some((balanceRes) => balanceRes.isError)
   return {
     accountBalance,
     connectedWalletsBalance,
     totalBalance,
-    isLoading
+    isLoading,
+    isError
+  }
+}
+
+function* getWalletBalances(wallets: Array<{ address: string; chain: Chain }>) {
+  const sdk = yield* call(getSDK)
+  const audiusBackend = yield* getContext<AudiusBackend>(
+    'audiusBackendInstance'
+  )
+  const queryClient = yield* getContext<QueryClient>('queryClient')
+  let totalBalance: AudioWei = AUDIO(0).value
+  for (const wallet of wallets) {
+    const balance = (yield* call(queryClient.fetchQuery, {
+      queryKey: getWalletAudioBalanceQueryKey({
+        address: wallet.address,
+        chain: wallet.chain,
+        includeStaked: true
+      }),
+      queryFn: async () =>
+        fetchWalletAudioBalance(
+          { sdk, audiusBackend },
+          { address: wallet.address, chain: wallet.chain, includeStaked: true }
+        )
+    })) as AudioWei | undefined
+    totalBalance = AUDIO(totalBalance + (balance ?? AUDIO(0).value)).value
   }
+  return totalBalance
+}
+
+export function* getAudioBalance() {
```

Comment:
it seems like both the callers of these generators used to use selectors, so I don't think we actually need to do the work to fetch the data and can probably pull the results from the queryClient? Wdyt?

We'd need to update the buyAudio transfer step to optimistically update the value (which i think we should do anyway to avoid race conditions) but I think that would eliminate the need for these generator helpers